yb_migrate/src/migration/postgres.go

```diff
@@ -236,49 +238,42 @@ func PgDumpExportDataOffline(ctx context.Context, source *utils.Source, exportDi
 	dataDirPath := exportDir + "/data"
 
 	tableListPatterns := createTableListPatterns(tableList)
-	// fmt.Printf("[Debug] createTableListPatterns = %+v\n", tableListPatterns)
+	log.Infof("createTableListPatterns = %s", tableListPatterns)
 
 	SSLQueryString := generateSSLQueryStringIfNotExists(source)
 
-	//using pgdump for exporting data in directory format
-	// pgdumpDataExportCommandArgsString := fmt.Sprintf(`pg_dump "postgresql://%s:%s@%s:%s/%s?%s" --data-only --compress=0 %s -Fd --file %s --jobs %d`, source.User, source.Password,
-	// 	source.Host, source.Port, source.DBName, SSLQueryString, tableListPatterns, dataDirPath, source.NumConnections)
-	pgdumpDataExportCommandArgsString := ""
+	// Using pgdump for exporting data in directory format.
+	cmd := ""
 	if source.Uri != "" {
-		pgdumpDataExportCommandArgsString = fmt.Sprintf(`pg_dump "%s" --no-blobs --data-only --compress=0 %s -Fd --file %s --jobs %d`, source.Uri, tableListPatterns, dataDirPath, source.NumConnections)
+		cmd = fmt.Sprintf(`pg_dump "%s" --no-blobs --data-only --compress=0 %s -Fd --file %s --jobs %d`, source.Uri, tableListPatterns, dataDirPath, source.NumConnections)
 	} else {
-		pgdumpDataExportCommandArgsString = fmt.Sprintf(`pg_dump "postgresql://%s:%s@%s:%d/%s?%s" --no-blobs --data-only --compress=0 %s -Fd --file %s --jobs %d`, source.User, source.Password,
+		cmd = fmt.Sprintf(`pg_dump "postgresql://%s:%s@%s:%d/%s?%s" --no-blobs --data-only --compress=0 %s -Fd --file %s --jobs %d`, source.User, source.Password,
 			source.Host, source.Port, source.DBName, SSLQueryString, tableListPatterns, dataDirPath, source.NumConnections)
 	}
-
-	// fmt.Printf("[Debug] Command: %s\n", pgdumpDataExportCommandArgsString)
-
-	pgdumpDataExportCommand := exec.CommandContext(ctx, "/bin/bash", "-c", pgdumpDataExportCommandArgsString)
-
-	var stderrBuffer bytes.Buffer
-	pgdumpDataExportCommand.Stderr = &stderrBuffer
-	pgdumpDataExportCommand.Stdout = &stderrBuffer
-
-	err := pgdumpDataExportCommand.Start()
-	fmt.Println("pg_dump for data export started")
+	log.Infof("Running: %s", cmd)
+	var buf bytes.Buffer
+	proc := exec.CommandContext(ctx, "/bin/bash", "-c", cmd)
+	proc.Stderr = &buf
+	proc.Stdout = &buf
+	err := proc.Start()
 	if err != nil {
-		fmt.Printf("%s\n%s\n", stderrBuffer.String(), err)
+		fmt.Printf("%s\n%s\n", buf.String(), err)
 		quitChan <- true
 		runtime.Goexit()
 	}
+	fmt.Println("Data export started.")
 	exportDataStart <- true
 
-	//Parsing the main toc.dat file in parallel
+	// Parsing the main toc.dat file in parallel.
 	go parseAndCreateTocTextFile(dataDirPath)
 
-	//Wait for pg_dump to complete before renaming of data files
-	err = pgdumpDataExportCommand.Wait()
+	// Wait for pg_dump to complete before renaming of data files.
+	err = proc.Wait()
 	if err != nil {
-		fmt.Printf("%s\n%s\n", stderrBuffer.String(), err)
+		fmt.Printf("%s\n%s\n", buf.String(), err)
```

Comment:
this will be important to log, as generally i have observed that error with pg_dump comes here at the end not at `proc.Start()`

Reply:
Done.